### PR TITLE
ability to comma separate platforms for agent names

### PR DESCRIPTION
**Pull request by scottctaylor12**
In the agents.yml file, rather than duplicate agent name lists for each platforms, lets add the ability to provide multiple platforms to a single list of agents by comma separating values.

### CODE REVIEW

File: app/service/contact_svc.py [hunk 1]
```diff
@@ -67,10 +67,26 @@ def __init__(self, agent_config):
...
-        self._file_names = agent_config['names']
+        self._file_names = self.collect_file_names(agent_config['names'])
         self._untrusted_timer = agent_config['untrusted_timer']
         self._bootstrap_instructions = agent_config['bootstrap_abilities']
 
+    def collect_file_names(self, names):
+        multiplatform = []
+        for platform, agents in names.items():
+            if "," in platform:
+                multiplatform.append(platform)
```
Comment (khyberspache) [inline app/service/contact_svc.py]: list comprehension

File: app/service/contact_svc.py [hunk 2]
```diff
@@ -67,10 +67,26 @@ def __init__(self, agent_config):
...
+
+        if multiplatform is not []:
+            for string in multiplatform:
```
Comment (khyberspache) [inline app/service/contact_svc.py]: another list comprehension that combines the names w/ multiplatform names

File: app/service/contact_svc.py [hunk 3]
```diff
@@ -67,10 +67,26 @@ def __init__(self, agent_config):
...
+                agents = names[string]
+                platforms = string.split(",")
+                for p in platforms:
+                    new = {p: agents}
+                    names.update(new)
+                del names[string]
+        return names
```
Comment (khyberspache) [inline app/service/contact_svc.py]: I think you can do this in 1 or 2 lines

File: app/service/contact_svc.py [hunk 4]
```diff
@@ -67,10 +67,26 @@ def __init__(self, agent_config):
         self._sleep_min = agent_config['sleep_min']
         self._sleep_max = agent_config['sleep_max']
         self._watchdog = agent_config['watchdog']
-        self._file_names = agent_config['names']
+        self._file_names = self.collect_file_names(agent_config['names'])
         self._untrusted_timer = agent_config['untrusted_timer']
         self._bootstrap_instructions = agent_config['bootstrap_abilities']
 
+    def collect_file_names(self, names):
+        multiplatform = []
```
Comment (privateducky) [inline app/service/contact_svc.py]: I think you can just change what calls self.-file-names instead of this

Comment (khyberspache) [reply app/service/contact_svc.py]: I don't think so - not if we are enabling comma-separated platform definitions (`linux,freebsd:`)

**Comment by privateducky** [comment]
no longer relevant :)

**Pull request closed without merging**